File: examples/java-maven-example/src/test/java/org/galaxio/performance/picatinny/TransactionCoverage.java
package org.galaxio.performance.picatinny;

import org.galaxio.gatling.javaapi.SimulationWithTransactions;
import org.galaxio.performance.picatinny.scenarios.TransactionScenario;

import static io.gatling.javaapi.core.CoreDsl.atOnceUsers;

public final class TransactionCoverage extends SimulationWithTransactions {
    {
        setUp(TransactionScenario.apply("Picatinny Transaction Coverage", "java-transaction-coverage").injectOpen(atOnceUsers(1)))
                .assertions(PerformanceSupport.noFailedRequests());
    }

    @Override
    public void before() {
        System.out.println("PICATINNY_TRANSACTION_COVERAGE_BEFORE_EXECUTED");
    }

    @Override
    public void after() {
        System.out.println("PICATINNY_TRANSACTION_COVERAGE_AFTER_EXECUTED");
    }
}

File: examples/kotlin-gradle-example/src/gatling/kotlin/org/galaxio/performance/picatinny/TransactionCoverage.kt
package org.galaxio.performance.picatinny

import io.gatling.javaapi.core.CoreDsl.atOnceUsers
import org.galaxio.gatling.javaapi.SimulationWithTransactions
import org.galaxio.performance.picatinny.scenarios.TransactionScenario

class TransactionCoverage : SimulationWithTransactions() {
    init {
        setUp(TransactionScenario.apply("Picatinny Transaction Coverage", "kotlin-transaction-coverage").injectOpen(atOnceUsers(1)))
            .assertions(PerformanceSupport.noFailedRequests())
    }

    override fun before() {
        println("PICATINNY_TRANSACTION_COVERAGE_BEFORE_EXECUTED")
    }

    override fun after() {
        println("PICATINNY_TRANSACTION_COVERAGE_AFTER_EXECUTED")
    }
}
